qa/rpc-tests/test_framework/mininode.py

```diff
@@ -1652,8 +1654,9 @@ def show_debug_msg(self, msg):
         self.log.debug(msg)
 
     def handle_connect(self):
-        self.show_debug_msg("MiniNode: Connected & Listening: \n")
-        self.state = "connected"
+        if self.state is not "connected":
```

Comment:
Should use `!=` instead of `is not` to avoid relying on string interning (also for two other string comparisons below).